packages/workbox-precaching/utils/getOrCreatePrecacheController.mjs

```diff
@@ -0,0 +1,24 @@
+/*
+  Copyright 2019 Google LLC
+
+  Use of this source code is governed by an MIT-style
+  license that can be found in the LICENSE file or at
+  https://opensource.org/licenses/MIT.
+*/
+
+import {PrecacheController} from '../PrecacheController.mjs';
+import '../_version.mjs';
+
+
+let precacheController;
+
+/**
+ * @return {PrecacheController}
+ * @private
+ */
+export const getOrCreatePrecacheController = () => {
+  if (!precacheController) {
+    precacheController = new PrecacheController();
```

Comment:
Are we losing the ability to customize the precache name?

C.f. https://github.com/GoogleChrome/workbox/blob/next/packages/workbox-precaching/_default.mjs#L29-L30 which will pass in the result of `cacheNames.getPrecacheName()` to the `PrecacheController` constructor.

Or maybe there's some other mechanism of customizing the name now?

Reply:
I don't think we're losing the ability to customize the name. Previously the code that instantiated the default `PrecacheController` looked like [this](https://github.com/GoogleChrome/workbox/blob/bab6be5/packages/workbox-precaching/_default.mjs#L29-L30), which passed in the current (or default) cache name, and the first line in `PrecacheController()` is [this](https://github.com/GoogleChrome/workbox/blob/bab6be502b3499e4d9bd12225716a298c6268395/packages/workbox-precaching/controllers/PrecacheController.mjs#L35), which uses the same method if no cache name was passed.

I believe in both before and after this PR, the way to change the cache name is still `workbox.core.setCacheNameDetails()`, and that needs to be done before using any `workbox-precaching` methods in both cases.

Reply:
Looking a bit closer at the code and the docs, it doesn't look like this is well documented or well tested. It's probably worth updating [this docs](https://developers.google.com/web/tools/workbox/guides/configure-workbox) to mention that setting of cache names needs to be done before loading any other modules.